pkg/api/validation/validation.go

```diff
@@ -1237,6 +1261,23 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, validateScaleIOVolumeSource(pv.Spec.ScaleIO, specPath.Child("scaleIO"))...)
 		}
 	}
+	if pv.Spec.LocalStorage != nil {
+		if numVolumes > 0 {
+			allErrs = append(allErrs, field.Forbidden(specPath.Child("localStorage"), "may not specify more than 1 volume type"))
+		} else {
+			numVolumes++
+			allErrs = append(allErrs, validateLocalStorageVolumeSource(pv.Spec.LocalStorage, specPath.Child("localStorage"))...)
+			// StorageClass is required
+			if len(pv.Spec.StorageClassName) == 0 {
+				allErrs = append(allErrs, field.Required(specPath.Child("storageClassName"), "LocalStorage volumes require a StorageClass"))
+			}
+			// hostname label is required
+			hostname, found := pv.Labels[metav1.LabelHostname]
```

Comment:
+1 for admission controller.

We should trust the admin that correct labels are set or the admission plugin is running. To be extra careful, SetUp() call of  volume plugin for LocalStorage could check that it runs on the right node before mount.